src/state/reducers/month-state/shifts-info.reducer.ts

```diff
@@ -16,13 +16,41 @@ import * as _ from "lodash";
 export function scheduleShiftsInfoReducerF(name: string) {
   return (
     state: ShiftInfoModel = scheduleDataInitialState.shifts,
-    action: ScheduleActionModel | ActionModel<WorkerActionPayload>
+    action:
+      | ScheduleActionModel
+      | ActionModel<WorkerActionPayload>
+      | ActionModel<Shift>
+      | ActionModel<Array<Shift>>
   ): ShiftInfoModel => {
     switch (action.type) {
       case createActionName(name, ScheduleActionType.ADD_NEW):
       case createActionName(name, ScheduleActionType.UPDATE):
         const data = (action.payload as ScheduleDataModel)?.shifts;
         return _.cloneDeep(data);
+
+      case ScheduleActionType.DELETE_SHIFT:
+        const { code } = action.payload as Shift;
+        Object.entries(state).forEach(([workerName, workersShifts]) => {
```

Comment:
Takie usuwanie powoduje usunięcie w kolejnym miesiącu, bo tam też leci update do bazy danych

Reply:
W tym miejscu to chyba nic nie powoduje.
Update jest robiony wcześniej.
Jedyny problem który widzę, to to, żę my usuwamy zmiany tylko ze stanu i nie usuwamy tego z BD
Dobrze myślę?

Reply:
A faktycznie masz rację, trochę się pośpieszyłem z wnioskami. Brakuje update'u do bazy. Z tym, że już przy samym update'cie trzeba uważać, żeby nie aktualizować kolejnego miesiąca.